tests/cupy_tests/test_cublas.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+
+import numpy
+
+import cupy
+from cupy import cublas
+from cupy import testing
+from cupy.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'dtype': ['float32', 'float64', 'complex64', 'complex128'],
+    'n': [10, 33, 100],
+    'bs': [None, 1, 10],
+    'nrhs': [None, 1, 10],
+}))
+@attr.gpu
+class TestBatchedGesv(unittest.TestCase):
+    _tol = {'f': 1e-5, 'd': 1e-12}
+
+    def _make_array(self, shape, alpha, beta):
+        a = testing.shaped_random(shape, cupy, dtype=self.r_dtype,
+                                  scale=alpha) + beta
+        return a
+
+    def _make_matrix(self, shape, alpha, beta):
+        a = self._make_array(shape, alpha, beta)
+        if self.dtype.char in 'FD':
+            a = a + 1j * self._make_array(shape, alpha, beta)
+        return a
+
+    def setUp(self):
+        self.dtype = numpy.dtype(self.dtype)
+        if self.dtype.char in 'fF':
+            self.r_dtype = numpy.float32
+        else:
+            self.r_dtype = numpy.float64
+        n = self.n
+        bs = 1 if self.bs is None else self.bs
+        nrhs = 1 if self.nrhs is None else self.nrhs
+        # Diagonally dominant matrix is used as it is stable
+        alpha = 2.0 / n
```

Comment:
This is neat!
I think we can use this in other places too

Reply:
@asi1024 pointed me out https://github.com/chainer/chainer/blob/7942e1cd425b29868ff872a5b9031f90a9bcda53/chainer/testing/matrix.py

Can you rewrite this test to use that routine to generate the matrix instead?
Sorry and thanks.

Reply:
Thanks for the pointer!  Certainly, it's safer to use SVD to make well-conditioned matrices.

Reply:
FYI
https://github.com/cupy/cupy/pull/4070

I will change this code once 4070 is merged